src/cobalt.rs

```diff
@@ -133,15 +133,35 @@ fn generate_doc(
 ) -> Result<()> {
     // Everything done with `globals` is terrible for performance.  liquid#95 allows us to
     // improve this.
-    let mut globals: liquid::value::Object = vec![
-        (
-            "site".into(),
-            liquid::value::Value::Object(context.site.clone()),
-        ),
-        global_collection,
-    ]
-    .into_iter()
-    .collect();
+    let mut globals: liquid::value::Object = if doc.front.pagination_compat {
+        let collections_posts: liquid::value::Object =
+            vec![("posts".into(), global_collection.1.clone())]
```

Comment:
Note to self: double check that what we are inserting in here is compatible with what we originally put in here

Reply:
@Geobert could you explain your theory of ops here?  It seems like the paginator is always enabled for all pages, regardless of what collection they are in (posts or pages) when
- I thought pagination is only supported (and so should error otherwise) in pages
- The goal was to be compatible while allowing people to opt-out.  It doesn't look like this provides an opt-out mechanism.

My thought for how this would have been implemented is to modify the `pages` specific defaults to set an active paginator over `all`.  That makes it easy for people to opt-out on a global level while then opting in on a per-page level.